<?xml version="1.0" encoding="utf-8"?>

<launch>
  <arg name="camera_type" value="$(env CAMERA)"/>
  <arg if="$(eval arg('camera_type') == 'astra')" name="input_image"        default="/camera/rgb/image_raw"/>
  <arg if="$(eval arg('camera_type') == 'astrapro')" name="input_image"        default="/camera/rgb/image_raw"/>
  <arg if="$(eval arg('camera_type') == 'usb')" name="input_image"        default="/camera/rgb/image_raw"/>
  <arg if="$(eval arg('camera_type') == 'csi')" name="input_image"        default="/camera/rgb/image_raw"/>
  <arg if="$(eval arg('camera_type') == 'intel')" name="input_image"      default="/camera/color/image_raw"/>

  <!-- Start yolov8 and ros wrapper -->
  <node pkg="rikibot_yolov8" type="yolov8_node.py" name="yolov8" output="screen" >
    <param name="weight_path"       value="$(find rikibot_yolov8)/weights/my_best.pt"/>
    <param name="image_topic"       value="$(arg input_image)" />
    <param name="pub_topic"         value="/yolov8/BoundingBox" />
    <param name="camera_frame"       value="camera"/>
    <param name="conf"              value="0.3" />
    <param name="use_cpu"           value="false" />
  </node>

</launch>
